src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/CvlRecordService.kt

```diff
@@ -0,0 +1,34 @@
+package uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service
+
+import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service.dates.ReleaseDateService
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service.prison.PrisonerSearchPrisoner
+
+@Service
+class CvlRecordService(
+  private val eligibilityService: EligibilityService,
+  private val releaseDateService: ReleaseDateService,
+) {
+
+  fun getCvlRecord(prisoner: PrisonerSearchPrisoner, areaCode: String): CvlCaseDto {
+    val recordList = getCvlRecords(listOf(prisoner), mapOf(prisoner.prisonerNumber to areaCode))
+    return recordList.find { it.nomisId == prisoner.prisonerNumber }!!
```

Comment:
This pattern of `find{...}!!` is used in a few places in regard to CvlRecords. There is no filtering of cases during either the CvlRecord building or the eligibility checking processes, and both are run based on PrisonerSearchPrisoners. This means that any case going in should have a record coming out, and it will be keyed by NOMIS ID.

Reply:
here you could just use `first()`